Sources/MapboxMaps/Foundation/Camera/CameraView.swift

```diff
@@ -166,83 +42,42 @@ internal class CameraView: UIView {
         self.isUserInteractionEnabled = false
-
-        // Sync default values from MBXMap
-        setFromValuesWithMapView()
     }
 
     internal required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    private func setFromValuesWithMapView() {
-        zoom = delegate.zoom
-        bearing = CGFloat(delegate.bearing)
-        pitch = delegate.pitch
-        padding = delegate.padding
-        centerCoordinate = delegate.centerCoordinate
-    }
-
-    internal func update() {
-
-        // Retrieve currently rendered camera
-        let currentCamera = delegate.cameraOptions
-
-        // Get the latest interpolated values of the camera properties (if they exist)
-        let targetCamera = localCamera.wrap()
-
-        // Apply targetCamera options only if they are different from currentCamera options
-        if currentCamera != targetCamera {
-
-            // Diff the targetCamera with the currentCamera and apply diffed camera properties to map
-            var diffedCamera = CameraOptions()
-
-            if targetCamera.zoom != currentCamera.zoom, let targetZoom = targetCamera.zoom, !targetZoom.isNaN {
-                diffedCamera.zoom = targetCamera.zoom
-            }
-
-            if targetCamera.bearing != currentCamera.bearing, let targetBearing = targetCamera.bearing, !targetBearing.isNaN {
-                diffedCamera.bearing = targetCamera.bearing
-            }
-
-            if targetCamera.pitch != currentCamera.pitch, let targetPitch = targetCamera.pitch, !targetPitch.isNaN {
-                diffedCamera.pitch = targetCamera.pitch
-            }
-
-            if targetCamera.center != currentCamera.center, let targetCenter = targetCamera.center, !targetCenter.latitude.isNaN, !targetCenter.longitude.isNaN {
-                diffedCamera.center = targetCamera.center
-            }
-
-            if targetCamera.anchor != currentCamera.anchor {
-                diffedCamera.anchor = targetCamera.anchor
-            }
+    internal func syncLayer(to cameraOptions: CameraOptions) {
+        if let zoom = cameraOptions.zoom {
+            layer.opacity = Float(zoom)
+        }
 
-            if targetCamera.padding != currentCamera.padding {
-                diffedCamera.padding = targetCamera.padding
-            }
+        if let bearing = cameraOptions.bearing {
+            layer.cornerRadius = CGFloat(bearing)
+        }
 
-            delegate.jumpTo(camera: diffedCamera)
+        if let centerCoordinate = cameraOptions.center {
+            layer.position = CGPoint(x: centerCoordinate.longitude, y: centerCoordinate.latitude)
         }
-    }
-}
 
-fileprivate extension CameraOptions {
+        if let padding = cameraOptions.padding {
+            layer.bounds = CGRect(x: padding.left,
+                                  y: padding.right,
+                                  width: padding.bottom,
+                                  height: padding.top)
```

Comment:
there is another bug that @neelmistry94 was looking into where we realized that bounds isn't a good choice for our purposes because the interpolated values are 'standardized' so that width and height are always positive. That breaks things when we try to pull values back out into a UIEdgeInsets.

Reply:
can be fixed separately though. there's no regression here relative to the previous implementation

Reply:
Yep, I'm aware of that issue. Lets fix it as part of a PR for that ticket specifically.